ui/src/app/components/analysis-modal/transforms.ts

```diff
@@ -618,9 +618,11 @@ const transformMeasurementValue = (conditionKeys: string[], value?: string): Mea
         };
     }
 
+    const safeValue = value.replace(/\bNaN\b/g, 'null');
+    
     let parsedValue;
     try {
-        parsedValue = JSON.parse(value);
+        parsedValue = JSON.parse(safeValue);
```

Comment:
If the value is `null` then we can skip the parse and  safely skip the value  or set it to 0